dev-docs/howto/vpn/on-prem-terraform/main.tf

```diff
@@ -0,0 +1,286 @@
+      next_hop_type          = "VirtualAppliance"
+      next_hop_in_ip_address = azurerm_network_interface.public_nic.private_ip_address
+    }
+  }
+}
+
+resource "azurerm_subnet_route_table_association" "route_table_association" {
+  subnet_id      = azurerm_subnet.subnet.id
+  route_table_id = azurerm_route_table.route_table.id
+}
+
+
+# Create network interface
+resource "azurerm_network_interface" "public_nic" {
+  name                = "public-nic"
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+
+  ip_configuration {
+    name                          = "my_nic_configuration"
+    subnet_id                     = azurerm_subnet.subnet.id
+    private_ip_address_allocation = "Dynamic"
+    public_ip_address_id          = azurerm_public_ip.pubIP.id
+  }
+}
+
+# Connect the security group to the network interface
+resource "azurerm_network_interface_security_group_association" "example" {
+  network_interface_id      = azurerm_network_interface.public_nic.id
+  network_security_group_id = azurerm_network_security_group.security_group.id
+}
+
+# Create virtual machine
+resource "azurerm_linux_virtual_machine" "public_vm" {
+  name                  = "public_vm"
+  location              = azurerm_resource_group.rg.location
+  resource_group_name   = azurerm_resource_group.rg.name
+  network_interface_ids = [azurerm_network_interface.public_nic.id]
+  size                  = "Standard_D2as_v5"
+
+  os_disk {
+    name                 = "disk_public_vm"
+    caching              = "ReadWrite"
+    storage_account_type = "Premium_LRS"
+  }
+
+  source_image_reference {
+    publisher = "Canonical"
+    offer     = "0001-com-ubuntu-server-jammy"
+    sku       = "22_04-lts-gen2"
+    version   = "latest"
+  }
+
+  computer_name  = "hostname"
+  admin_username = "azureadmin"
+
+  admin_ssh_key {
+    username   = "azureadmin"
+    public_key = tls_private_key.ssh_key.public_key_openssh
+  }
+
+  boot_diagnostics {
+  }
+
+  user_data = base64encode(<<EOF
+#!/bin/bash
+set -x
+
+apt-get update
+apt-get install strongswan-charon strongswan-swanctl -y
+
+
+cat <<EOT >> /etc/strongswan.d/charon-logging.conf
+charon {
+  filelog {
+    stderr {
+      time_format = %b %e %T
+      ike_name = yes
+      default = 1
+      ike = 2
+      flush_line = yes
+    }
+  }
+}
+EOT
+
+
+cat <<EOT >> /etc/swanctl/conf.d/constellation.conf
+connections {
+  gw-gw {
+    remote_addrs = ${var.remote_addr}
+
+    local {
+        auth = psk
+    }
+    remote {
+        auth = psk
+    }
+    children {
+        net-net {
+          local_ts  = ${var.local_ts}
+          remote_ts = ${join(",", var.remote_ts)}
+
+          start_action = trap
+        }
+    }
+  }
+}
+
+secrets {
+  ike {
+    secret = ${var.ike_psk}
+  }
+}
+EOT
+
+cat <<EOT >> /entrypoint.sh
+#!/bin/sh
+
+while ! swanctl --stats > /dev/null 2> /dev/null; do
+  sleep 1
+done
+swanctl --load-all
+
+wait
```

Comment:
This looks like you're starting Charon via systemd. In that case, you can remove the logging configuration and consume from `journalctl -u strongswan`.

Reply:
Actually, I think we'd not need to do any `swanctl` if we were using the systemd variant.

Reply:
There is no output for `journalctl -u strongswan`. I also don't install the `charon-systemd` package. While I tried for a few minutes to use the systemd variant, I gave up when this variant worked.

Reply:
That makes perfect sense. But where is Charon being started here, I wonder?

Reply:
I'd imagine that it is stared when loading the config via swanctl.